### ISSUE TITLE
DEV/RC2: Sign on fails under some circumstances (maybe StartCom SSL issue)

### ISSUE THREAD

**Issue by DemoFreak** [nextcloud/android#1359]
### Actual behaviour
When trying to log in the applications screen remains blank forever just after tapping on right arrow icon next to the server address.

### Expected behaviour
Login screen appears for input of credentials.
 
### Steps to reproduce
1. install Dev/RC version
2. open, skip to the input of server address
3. input server address and tap right-arrow icon

### Comment
This behavior occurs with my Nextcloud server at the web address where the Startcom SSL certificate is issued to. The same server is accessible under another address (where the certificate isn't issued to). When trying to log into the server using this address a warning message appears asking me if I want to trust this certificate anyway. After affirmation all seems fine, the app works as expected (asking for grant access and login credentials).
When I go back at this point and try to use a the "right web address" instead a second time, this time it works also with this address.

### Environment data
Android version: 6.0.1
Device model: Galaxy Note4
Stock or customized system: stock
Nextcloud app version: 2.0 RC2
Nextcloud server version: 12.0.1

#### Web server access log
```
146.44.44.113 - - [17/Aug/2017:00:23:47 +0200] "GET /status.php HTTP/1.1" 200 5535 "-" "Mozilla/5.0 (Android) ownCloud-android/2.0.0RC2"
146.44.44.113 - - [17/Aug/2017:00:23:47 +0200] "HEAD /remote.php/webdav/ HTTP/1.1" 401 1206 "-" "Mozilla/5.0 (Android) ownCloud-android/2.0.0RC2"
```

#### Web server access log
empty

#### Nextcloud log (data/nextcloud.log)
First try:
```
{"reqId":"IsFEPqwRWq3EhGAF5FkN","level":0,"time":"2017-08-16T22:23:47+00:00","remoteAddr":"146.44.44.113","user":"--","app":"webdav","method":"HEAD","url":"\/remote.php\/webdav\/","message":"Exception: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"
Message\":\"No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured, No 'Authorization: Bearer' header found. Either the client didn't send one, or the server is mis-configured\",\"Code\":0,\"Trace\":\"#0 [internal funct
ion]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Arr
ay, Array)\\n#2 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\
\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(71): Sabre\\\\DAV\\
\\Server->exec()\\n#5 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/remote.php(162): require_once('\\\/srv\\\/www\\\/client...')\\n#6 {main}\",\"File\":\"\\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/
Plugin.php\",\"Line\":168}","userAgent":"Mozilla\/5.0 (Android) ownCloud-android\/2.0.0RC2","version":"12.0.1.5"}
```

Second try (when it works):
```
{"reqId":"UCboiM31NNVklOuOUL4S","level":0,"time":"2017-08-16T22:46:02+00:00","remoteAddr":"146.44.44.113","user":"--","app":"webdav","method":"HEAD","url":"\/remote.php\/webdav\/","message":"Exception: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Message\":\"No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured, No 'Authorization: Bearer' header found. Either the client didn't send one, or the server is mis-configured\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(71): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/remote.php(162): require_once('\\\/srv\\\/www\\\/client...')\\n#6 {main}\",\"File\":\"\\\/srv\\\/www\\\/clients\\\/client4\\\/web27\\\/web\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":168}","userAgent":"Mozilla\/5.0 (Android) ownCloud-android\/2.0.0RC2","version":"12.0.1.5"}
{"reqId":"31VZfqxHIzf4lZBdSiM2","level":3,"time":"2017-08-16T22:46:03+00:00","remoteAddr":"146.44.44.113","user":"--","app":"PHP","method":"GET","url":"\/index.php\/login\/flow","message":"Undefined index: oauthState at \/srv\/www\/clients\/client4\/web27\/web\/core\/templates\/loginflow\/authpicker.php#38","userAgent":"Samsung SM-N910F","version":"12.0.1.5"}
{"reqId":"31VZfqxHIzf4lZBdSiM2","level":0,"time":"2017-08-16T22:46:03+00:00","remoteAddr":"146.44.44.113","user":"--","app":"core","method":"GET","url":"\/index.php\/login\/flow","message":"Scss is disabled for \/srv\/www\/clients\/client4\/web27\/web\/core\/css\/server.scss, ignoring","userAgent":"Samsung SM-N910F","version":"12.0.1.5"}
{"reqId":"31VZfqxHIzf4lZBdSiM2","level":0,"time":"2017-08-16T22:46:03+00:00","remoteAddr":"146.44.44.113","user":"--","app":"core","method":"GET","url":"\/index.php\/login\/flow","message":"Scss is disabled for \/srv\/www\/clients\/client4\/web27\/web\/core\/css\/share.scss, ignoring","userAgent":"Samsung SM-N910F","version":"12.0.1.5"}
{"reqId":"I9aIsrQQB93cLlL9LbKp","level":0,"time":"2017-08-16T22:46:04+00:00","remoteAddr":"146.44.44.113","user":"--","app":"core","method":"GET","url":"\/index.php\/apps\/theming\/styles?v=0","message":"apps\/theming\/css\/theming.scss compiled and successfully cached","userAgent":"Samsung SM-N910F","version":"12.0.1.5"}
```

**Comment by tobiasKaminsky** [comment]
As you offered in IRC, can you please write an email to android at nextcloud dot com with a test account (and link to that issue).
Thank you for reporting this and helping fixing :-)

**Comment by DemoFreak** [comment]
Done.

**Comment by tobiasKaminsky** [comment]
Thank you.
I tested it and it indeed seems to be a cert issue on your first host (cloud.s....) whereas the second one is working fine for me (after a warning).
To get the first one working we would have to weaken the security, which we will not do.
I suggest to use LetsEncrypt for your non working host, which is easy to set up.

**Comment by DemoFreak** [comment]
I agree regarding the fact that I should replace StartCom by LetsEncrypt. It's on the agenda since weeks. ;-)

But IMHO it's clearly a bug, that the app simply shows a blank screen without giving the user a hint what's going wrong. Especially as it works fine without any hassle after trying another host in between.

**Comment by tobiasKaminsky** [comment]
I narrowed it down, a fix is in #1361.
It will be soon in dev version.

**Comment by DemoFreak** [comment]
Dev version 20170817 fixes this for me.
Thanks. :)
